67_RayQueryGeometry/app_resources/render.comp.hlsl

```diff
@@ -0,0 +1,160 @@
+#include "common.hlsl"
+
+#include "nbl/builtin/hlsl/spirv_intrinsics/core.hlsl"
+
+struct SPushConstants
+{
+    SGeomBDA geom[OT_COUNT];
+};
+
+[[vk::push_constant]] SPushConstants pc;
+
+[[vk::binding(0, 0)]]
+RaytracingAccelerationStructure topLevelAS;
+
+[[vk::binding(1, 0)]]
+cbuffer CameraData
+{
+    SCameraParameters params;
+};
+
+[[vk::binding(2, 0)]] RWTexture2D<float4> outImage;
+
+float3 unpackNormals3x10(uint32_t v)
+{
+    // host side changes float32_t3 to EF_A2B10G10R10_SNORM_PACK32
+    float3 n = (float3)0;
+    const uint mask = 0x3FF;    // 10 bits
+    n.x = float(asint((v >> 20) & mask));
+    n.y = float(asint((v >> 10) & mask));
+    n.z = float(asint((v >> 0) & mask));
+    return n;
+}
+
+// How the normals are packed
+// cube                     uint32_t - 8, 8, 8, 8
+// sphere, cylinder, arrow  uint32_t - 10, 10, 10, 2
+// rectangle, disk          float3
+// cone                     uint32_t - 10, 10, 10, 2
+// icosphere                float3
+static const uint indexTypes[OT_COUNT]      = { 0, 1, 0, 0, 2, 0, 0, 1 };
+static const uint vertexStrides[OT_COUNT]   = { 24, 28, 28, 32, 32, 28, 20, 32 };
```

Comment:
everything should probably live together in one struct, vertex buffer offset, index buffer offset, geometry type, index  type, vertex strides